packages/legacy-events/SyntheticEvent.js

```diff
@@ -152,71 +154,79 @@ Object.assign(SyntheticEvent.prototype, {
    * won't be added back into the pool.
    */
   persist: function() {
-    this.isPersistent = functionThatReturnsTrue;
+    // Modern event system doesn't use pooling.
+    if (!enableModernEventSystem) {
```

Comment:
We'd probably want to add a warning on using persist(), as it'll just be dead code in apps otherwise.